lib/pbench/server/auth/__init__.py

```diff
@@ -21,21 +22,180 @@ def __init__(self, http_status: int, message: str = None):
         self.message = message if message else HTTPStatus(http_status).phrase
 
     def __repr__(self) -> str:
-        return f"Oidc error {self.http_status} : {str(self)}"
+        return f"OIDC error {self.http_status} : {str(self)}"
 
     def __str__(self) -> str:
         return self.message
 
 
-class OpenIDCAuthenticationError(OpenIDClientError):
+class OpenIDTokenInvalid(Exception):
     pass
 
 
+class Connection:
+    """Helper connection class for use by an OpenIDClient instance."""
+
+    def __init__(
+        self,
+        server_url: str,
+        headers: Optional[Dict[str, str]] = None,
+        verify: bool = True,
+    ):
+        self.server_url = server_url
+        self.headers = CaseInsensitiveDict({} if headers is None else headers)
+        self.verify = verify
+        self._connection = requests.Session()
+
+    def _method(
+        self,
+        method: str,
+        path: str,
+        data: Union[Dict, None],
+        headers: Optional[Dict] = None,
+        **kwargs,
+    ) -> requests.Response:
+        """Common frontend for the HTTP operations on OIDC client connection.
+
+        Args:
+            method : The API HTTP method
+            path : Path for the request.
+            data : Json data to send with the request in case of the POST
+            kwargs : Additional keyword args
+
+        Returns:
+            Response from the request.
+        """
+        final_headers = self.headers.copy()
+        if headers is not None:
+            final_headers.update(headers)
+        url = urljoin(self.server_url, path)
+        kwargs = dict(
+            params=kwargs,
+            data=data,
+            headers=final_headers,
+            verify=self.verify,
+        )
```

Comment:
Even if the data is `None` that shouldn't affect the request right? The `requests.get` can have `data=None`.

Reply:
That's correct; in fact, a `GET` **should** have no data...in which case, it would be better to omit the keyword (although, in practice, it presumably makes no difference).